### DISCUSSION TITLE
Can Linux dynamically allocate cores to either Linux or other RTOS in AMP scenario?

### DISCUSSION THREAD

**Discussion by josh25219** [polarfire-soc/polarfire-soc-documentation#180] [Bare metal embedded software] [answered]
Hi,

I have a customer who has asked if they were running in an AMP scenario with Linux running on 1 or more cores if Linux would be able to dynamically control how many cores were allocated to Linux vs how many were associated with the other RTOS.

I know we don't support this currently.   Just wondering if this has been discussed for the future, or if its even possible from a hardware perspective?

THanks,
Josh

**Comment by hughbreslin** [comment, marked as answer]
HI @josh25219 

technically this is possible but its not something we're actively discussing or planning for. Why do they want to dynamically run the RTOS?
